### PR TITLE
ImportSignupFlow: Add confirm-import step

### PR DESCRIPTION
**Pull request by spen**
> Note: DO NOT MERGE - we need to make sure that, once in Calypso-proper, folk don't see the site preview there as well as in signup - this is being worked on currently.
> Note: The first 4 commits on this branch have been cherry-picked from #31588 (✅),  #31705 (✅) and #31643 (❌). Those 3 PRs should be merged first to help minimise the size of this PR, making for easier reviewing.

### Summary

This PR adds an extra step to the import signup flow which shows a preview of the targeted site and asks the customer to either confirm their ownership of the site, or, if it's not their site, go back to the previous step to try a different URL.

Screenshot 2019-03-25 at 12 10 31

### Testing Instructions

First, let's make sure that there are no regressions in our validation:

1. Apply this branch, then visit `/start/import/from-url`
1. Try submitting a random string  that is not a valid url (example: `not-a-valid-url`)
1. You should see an error notice, stating: "Please enter a valid URL."
1. Try submitting a url that is a valid site, but isn't importable (example: `google.com`)
1. You should see an error notice, stating: "We're not able to reach that site at this time, or it's not compatible with our URL importer."

Then, let's test the new step:

1. Try submitting a url that is a valid wix site (example: `https://smt593.wixsite.com/wowz`)
1. You should see that the button shows a busy state and a notice is shown telling you that we're checking to see if it's importable
1. Since the site is importable, we should then be taken directly to our new `confirm-import` step, where we should see the screen shown in the above screenshot.
1. Hit the negative button
1. You should be taken back to the url input step and it should function as before
1. Submit the same URL again, but this time hit the positive button to carry on with the signup flow
1. If signed out (on an incognito browser) you should be taken to the username & password step.
1. Follow this through until you land in Calypso. You'll notice that we're on the Wix import screen there, ready to confirm our import (future work will change this, so that we don't need to confirm the site preview twice)

### CODE REVIEW

**Comment by spen** [comment]
Notes to self: Needs keyboard/accessibility attention... should we be focused on the confirmation button when we come to this step?

**Comment by dezzie** [comment]
@spen regarding the focus of the confirmation button on this step. I think that's the best choice, yeah.

A screenreader would start reading the URL for the site, a person could choose to go back from that point. To mitigate the possibility that the URL is not pronounceable, or takes too long to spell out, we could consider putting the site name instead of the URL in that message.

Either way it's fine by me, I think this is good enough to move along.

**Comment by creativecoder** [comment]
Note that site mockup component was removed in https://github.com/Automattic/wp-calypso/pull/32135 and replaced with an iframe. To get this working again we'll need to re-add the SiteMockup component (https://github.com/Automattic/wp-calypso/pull/32135/files#diff-e355c99abf5e27412862cefaede316aa). It can be further simplified to be just a border that wraps an image to make it look like a browser window.

**Comment by creativecoder** [comment]
I've just rebased this against master and fixed conflicts.

File: client/components/site-mockup/index.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+					<rect x="28" width="10" height="10" rx="5" />
+				</g>
+			</svg>
+			<span className="site-mockup__chrome-label">{ translate( 'Website Preview' ) }</span>
+		</div>
+	);
+}
+
+const SiteMockup = React.memo(
+	( { className, children, siteStyle, siteType, size = 'mobile', tagline, title } ) => (
```
Comment (creativecoder) [inline client/components/site-mockup/index.jsx]: @spen Given the use of `SiteMockup` here is much simpler than how this component was previously used, what do you think of removing `siteStyle`, `siteType`, `tagline`, and `title` and keep this just as the "browser chrome" around an image?

Comment (spen) [reply client/components/site-mockup/index.jsx]: That makes sense to me! We could cut down on a lot of this without those props :)

**Comment by spen** [comment]
> Taking ownership of this, I'm going to modify a few things to bring it closer in line with recent design work for this flow.

Awesome, thanks so much Grant!
I can help try to get it merged before the weekend, else we can clear up this and some other bits next week :)

**Comment by creativecoder** [comment]
In the most recent iteration of import in signup designs, we're showing a much simplified preview of the site to be imported (favicon, title, and url only) with information on what will be imported. Sadly I'm closing this one.

**Pull request closed without merging**